新規登録、ログイン機能の実装（users.devise)

ozeryu7 opened the pull request:
#何か
deviseを用いたuser新規登録、ログイン機能を実装します。

msgyu commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,11 @@
+
+  def new
+    @user = User.new
+  end
```
今回はログイン/新規登録機能だけが必要なので、これらのアクションは不要となります。

ozeryu7 replied:
app/controllers/users_controller.rb 削除完了しました。

ozeryu7 committed:
user.modelのバリデーションを修正

下記のようにバリデーションを修正
・メールアドレスに全角文字が含まれていないこと
・パスワードは8文字以上32文字以下であること
・パスワードは半角英字と半角数字のいずれとも含まれていること

ozeryu7 committed:
gemファイルにエラーメッセージのgemを追加

rails-i18nを追加

msgyu committed:
users_controllerのアクションを削除

deviseのコントローラーを使用するため、users_controller.rbの記述は削除した。

ozeryu7 committed:
新規登録時に日本語でエラーメッセージが出力されるようにconfig_application、config_localesに追記

i18nを使用して日本語化

ozeryu7 committed:
Merge branch 'origin/user.devise' of https://github.com/ozeryu7/tenkue-board-r006 into origin/user.devise

ozeryu7 committed:
create devise.ja.yml でエラーメッセージを日本語化

新規登録時のエラーメッセージを日本語化しました。

ozeryu7 committed:
ログイン時のパスワードに半角英数字のみ許可する修正

ログインのバリデーションに不具合があったので修正

ozeryu7 committed:
application.scss、application.htmlに新規登録、ログイン、ログアウトの際のフラッシュメッセージの実装

コード入れましたがどうもうまく実装できないです。

msgyu commented on app/assets/javascripts/posts.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
不要なファイルの削除

ozeryu7 replied:
app/assets/javascripts/posts.coffee　削除完了しました。

msgyu commented on app/assets/javascripts/users.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
不要なファイルの削除

ozeryu7 replied:
app/assets/javascripts/users.coffee　削除完了しました。

msgyu commented on app/assets/stylesheets/posts.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the posts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
不要なファイルの削除

ozeryu7 replied:
app/assets/stylesheets/posts.scss　削除完了しました。

msgyu commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
不要なファイルの削除

ozeryu7 replied:
app/assets/stylesheets/users.scss　削除完了しました。

msgyu commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,3 @@
+class UsersController < ApplicationController
+  
+end
```
不要なコントローラーの削除

ozeryu7 replied:
app/controllers/users_controller.rb　削除完了しました。

ozeryu7 committed:
不要なファイル5個の削除 posts.coffee, users.coffee, posts.scss, users.scss, users.controller

nousi commented on db/migrate/20201024140953_create_posts.rb:
```diff
@@ -1,7 +1,6 @@
 class CreatePosts < ActiveRecord::Migration[5.2]
   def change
-    create_table :posts do |t|
-
+    create_table :postss do |t|
       t.timestamps
     end
   end
```
このファイルは不要なので、削除してください。
削除後は rake db:reset を実施してください

msgyu committed:
Merge branch 'master' into origin/user.devise

msgyu committed:
.noticeのcssをheader.scssに記述

application.scssから移行

msgyu committed:
_header.html.erbを1つだけ残し、不要なファイルを削除

msgyu committed:
_header.html.erbをsharedフォルダに移行

msgyu commented:
LGTM

ozeryu7 commented:
LGTMです。

masato-sa93 reviewed (approved):
LGTMですー！
一発はさすがです！

msgyu committed:
Merge branch 'master' into origin/user.devise
